--- kchart/CMakeLists.txt
+++ kchart/CMakeLists.txt
@@ -74,7 +74,7 @@
 
 kde4_add_kdeinit_executable( kchart ${kchart_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_kchart  ${KDE4_KDECORE_LIBS} komain )
+target_link_libraries(kdeinit_kchart  ${KDE4_KDECORE_LIBS} komain ${QT_QTGUI_LIBRARY} )
 
 install(TARGETS kdeinit_kchart  DESTINATION ${LIB_INSTALL_DIR})
 
--- kchart/shape/CMakeLists.txt
+++ kchart/shape/CMakeLists.txt
@@ -28,7 +28,7 @@
 )
 
 kde4_add_library(chartshapelib SHARED ${chartshape_LIB_SRCS})
-target_link_libraries(chartshapelib kdchart kochart flake)
+target_link_libraries(chartshapelib kdchart kochart flake komain koodf kstore ${KDE4_KPARTS_LIBS} ${QT_QT3SUPPORT_LIBRARY} )
 set_target_properties(chartshapelib PROPERTIES VERSION ${GENERIC_KOFFICE_LIB_VERSION} SOVERSION ${GENERIC_KOFFICE_LIB_SOVERSION} )
 
 install(TARGETS chartshapelib DESTINATION ${LIB_INSTALL_DIR})
