From 3d94eba26d12c3505102f7ac18bb67ce3275f213 Mon Sep 17 00:00:00 2001
From: Bernd Steinhauser <gentoo@bernd-steinhauser.de>
Date: Sat, 19 Apr 2008 00:35:00 +0200
Subject: [PATCH] Add more headers to install routine.

---
 workspace/libs/plasma/CMakeLists.txt |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/workspace/libs/plasma/CMakeLists.txt b/workspace/libs/plasma/CMakeLists.txt
index e98c1af..9a74bd0 100644
--- a/workspace/libs/plasma/CMakeLists.txt
+++ b/workspace/libs/plasma/CMakeLists.txt
@@ -135,6 +135,9 @@ install(FILES
 install(FILES
     widgets/icon.h
     widgets/webcontent.h
+	widgets/lineedit.h
+	widgets/pushbutton.h
+	widgets/widget.h
     widgets/meter.h
     widgets/signalplotter.h
     DESTINATION ${INCLUDE_INSTALL_DIR}/plasma/widgets)
-- 
1.5.4.5

