diff -Naur tar-1.19/configure.ac tar-1.19-patched/configure.ac
--- tar-1.19/configure.ac	2007-10-10 12:59:28.000000000 +0200
+++ tar-1.19-patched/configure.ac	2007-12-09 17:23:33.000000000 +0100
@@ -232,6 +232,11 @@
                     ;;
                esac])
 
+AC_ARG_ENABLE(7zip,
+              AC_HELP_STRING([--enable-7zip],
+               [Enable 7zip support.]),
+              AC_DEFINE([USE_7Z], [], [Enable use of 7zip]))						  
+
 AC_SUBST(BACKUP_SED_COND)
 if date +%Y-%m-%d 2>/dev/null >&2; then
 	BACKUP_SED_COND='/^\#ELSE_DATE_FORMAT_OK/,/^\#ENDIF_DATE_FORMAT_OK/d;/^\#IF_DATE_FORMAT_OK/d'
diff -Naur tar-1.19/src/tar.c tar-1.19-patched/src/tar.c
--- tar-1.19/src/tar.c	2007-09-26 23:36:58.000000000 +0200
+++ tar-1.19-patched/src/tar.c	2007-12-09 17:18:04.000000000 +0100
@@ -575,7 +575,11 @@
   {"label", 'V', N_("TEXT"), 0,
    N_("create archive with volume name TEXT; at list/extract time, use TEXT as a globbing pattern for volume name"), GRID+8 },
   {"bzip2", 'j', 0, 0,
-   N_("filter the archive through bzip2"), GRID+8 },
+   N_("filter the archive through bzip2"), GRID+8 },   
+#ifdef USE_7Z
+  {"7zg", 'e', 0, 0,
+   N_("filter the archive through 7zg"), GRID+8 },
+#endif
   {"gzip", 'z', 0, 0,
    N_("filter the archive through gzip"), GRID+8 },
   {"gunzip", 0, 0, OPTION_ALIAS, NULL, GRID+8 },
@@ -1310,6 +1314,12 @@
       set_use_compress_program_option ("bzip2");
       break;
 
+#ifdef USE_7Z
+    case 'e':
+      set_use_compress_program_option ("7zg");
+      break;
+#endif
+
     case 'k':
       /* Don't replace existing files.  */
       old_files_option = KEEP_OLD_FILES;
