--- kivio/src/part/CMakeLists.txt
+++ kivio/src/part/CMakeLists.txt
@@ -11,7 +11,7 @@
 
 kde4_add_library(kivioprivate SHARED ${kivioprivate_LIB_SRCS})
 
-target_link_libraries(kivioprivate  ${KDE4_KDECORE_LIBS} koguiutils flake kopageapp)
+target_link_libraries(kivioprivate  ${KDE4_KDECORE_LIBS} koguiutils flake kopageapp komain ${KDE4_KPARTS_LIBS} ${QT_QT3SUPPORT_LIBRARY} )
 
 set_target_properties(kivioprivate PROPERTIES VERSION ${GENERIC_KOFFICE_LIB_VERSION} SOVERSION ${GENERIC_KOFFICE_LIB_SOVERSION} )
 install(TARGETS kivioprivate  DESTINATION ${LIB_INSTALL_DIR})
@@ -32,7 +32,7 @@
 
 kde4_add_kdeinit_executable( kivio ${kivio_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_kivio  ${KDE4_KDECORE_LIBS} komain )
+target_link_libraries(kdeinit_kivio  ${KDE4_KDECORE_LIBS} komain ${KDE4_KPARTS_LIBS} )
 
 install(TARGETS kdeinit_kivio DESTINATION ${LIB_INSTALL_DIR})
 
